com.microsoft.mrtk.spatialmanipulation/BoundsControl/Visuals/SpatialManipulationReticle.cs

```diff
@@ -1,70 +1,97 @@
-//Copyright(c) Microsoft Corporation.
+// Copyright(c) Microsoft Corporation.
 // Licensed under the MIT License.
 
 using System;
 using UnityEngine;
+using UnityEngine.XR.Interaction.Toolkit;
 
 namespace Microsoft.MixedReality.Toolkit.SpatialManipulation
 {
-    public class SpatialManipulationReticle : MonoBehaviour
+    /// <summary>
+    /// A reticle used to visualize spatial manipulation capabilities when hovering over a bounding box handle.
+    /// The reticle is oriented in relation to the bounding box, to indicate the direction for rotation or scaling.
+    /// </summary>
+    public class SpatialManipulationReticle : MonoBehaviour, IVariableReticle
     {
-
-        [SerializeField]
-        [Tooltip("The type of the reticle visuals. Scale or Rotate.")]
-        private SpatialManipulationReticleType reticleType;
+        /// <summary>
+        /// The type of the reticle visuals. Scale or Rotate.
+        /// </summary>
+        [field: SerializeField, Tooltip("The type of the reticle visuals. Scale or Rotate.")]
+        public SpatialManipulationReticleType ReticleType { get; set; }
 
         private Quaternion worldRotationCache;
 
+        /// <summary>
+        /// Called by once per frame by <see cref="MRTKRayReticleVisual"/> from its UpdateReticle.
+        /// Rotates the cursor reticle based on the hovered or selected handle's position relative to the box visuals. 
+        /// </summary>
+        public void UpdateVisuals(VariableReticleArgs args)
+        {
+            GameObject customReticle = args.Reticle;
```

Comment:
I'm pretty sure that this.GameObject will equal args.Reticle  .... so you don't need args.Reticle.